### PR TITLE
net/usrsock: Add flags to sendto/recvfrom in usrsock

### PR DESCRIPTION
**Pull request by SPRESENSE**
## Summary

Add flags argument to sendto/recvfrom interface in usrsock.

## Testing

Tested by spresense:wifi

### CODE REVIEW

File: net/usrsock/usrsock_sendto.c [hunk 1]
```diff
@@ -172,6 +172,7 @@ static int do_sendto_request(FAR struct usrsock_conn_s *conn,
 
   req.head.reqid = USRSOCK_REQUEST_SENDTO;
   req.usockid = conn->usockid;
+  req.flags = flags;
```
Comment (xiaoxiang781216) [inline net/usrsock/usrsock_sendto.c:175]: Please submit the related patch to:
apps/examples/usrsocktest/
apps/wireless/gs2200m/
apps/netutils/usrsock_rpmsg/

File: net/usrsock/usrsock_sendto.c [hunk 2]
```diff
@@ -376,9 +377,13 @@ ssize_t usrsock_sendto(FAR struct socket *psock, FAR const void *buf,
           goto errout_unlock;
         }
 
+      /* MSG_DONTWAIT is only use in usrsock. */
+
+       flags &= ~MSG_DONTWAIT;
```
Comment (xiaoxiang781216) [inline net/usrsock/usrsock_sendto.c:382]: Should we need handle DONTWAIT like NONBLOCK flag?

File: net/usrsock/usrsock_recvfrom.c [hunk 3]
```diff
@@ -397,9 +399,13 @@ ssize_t usrsock_recvfrom(FAR struct socket *psock, FAR void *buf, size_t len,
 
       usrsock_setup_datain(conn, inbufs, ARRAY_SIZE(inbufs));
 
+      /* MSG_DONTWAIT is only use in usrsock. */
+
+      flags &= ~MSG_DONTWAIT;
```
Comment (xiaoxiang781216) [inline net/usrsock/usrsock_recvfrom.c:404]: Should we need handle DONTWAIT like NONBLOCK flag?

File: net/usrsock/usrsock_sendto.c (hunk 1 above)
Comment (SPRESENSE) [reply net/usrsock/usrsock_sendto.c:175]: Okay, I'll push them later.

File: net/usrsock/usrsock_sendto.c (hunk 2 above)
Comment (SPRESENSE) [reply net/usrsock/usrsock_sendto.c:382]: yes, we need it.

File: net/usrsock/usrsock_recvfrom.c (hunk 3 above)
Comment (SPRESENSE) [reply net/usrsock/usrsock_recvfrom.c:404]: Yes, we need it.

**Comment by jerpelea** [comment]
please add comiit message with the commit description

**Comment by SPRESENSE** [comment]
@jerpelea I added the commit message. Could you please review this?

**Comment by jerpelea** [comment]
@SPRESENSE Please send the complementary patches to the apps git

**Comment by xiaoxiang781216** [comment]
LGTM.

**Pull request merged**